src/unstated.js

```diff
@@ -139,3 +139,31 @@ export function Provider(props: ProviderProps) {
     </StateContext.Consumer>
   );
 }
+
+type ContainersMap = {
+  [string]: *
+};
+
+export const withSubscription = (containersMap: ContainersMap) => (
+  WrappedComponent: ComponentType<*>
+) => (props: *) => {
+  const containerKeys = Object.keys(containersMap);
+  const containers = containerKeys.map(
+    containerKey => containersMap[containerKey]
+  );
+
+  return (
+    <Subscribe to={containers}>
+      {(...containerInstances) => {
+        const containerProps = containerInstances.reduce(
+          (accumulator, instance, index) =>
+            Object.assign({}, accumulator, {
```

Comment:
You create this object out of the void anyway (new object gets passed as initial accumulator), so I'd suggest just assigning **onto** accumulator in this iterating function - intermediate objects won't have to be GCed. As this mutation is local it shouldn't lead to any problems.